aws/resource_aws_mwaa_environment_test.go

```diff
@@ -0,0 +1,770 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/mwaa"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func init() {
+	resource.AddTestSweepers("aws_mwaa_environment", &resource.Sweeper{
+		Name: "aws_mwaa_environment",
+		F:    testSweepMwaaEnvironment,
+	})
+}
+
+func testSweepMwaaEnvironment(region string) error {
+	client, err := sharedClientForRegion(region)
+	if err != nil {
+		return fmt.Errorf("error getting client: %s", err)
+	}
+	conn := client.(*AWSClient).mwaaconn
+
+	listOutput, err := conn.ListEnvironments(&mwaa.ListEnvironmentsInput{})
+	if err != nil {
+		if testSweepSkipSweepError(err) || isAWSErr(err, "InternalFailure", "") {
+			log.Printf("[WARN] Skipping MWAA Environment sweep for %s: %s", region, err)
+			return nil
+		}
+		return fmt.Errorf("Error retrieving MWAA Environment: %s", err)
+	}
+	for _, environment := range listOutput.Environments {
+		name := aws.StringValue(environment)
+		r := resourceAwsMwaaEnvironment()
+		d := r.Data(nil)
+		d.SetId(name)
+
+		err := r.Delete(d, client)
+		if err != nil {
+			log.Printf("[ERROR] Failed to delete MWAA Environment %s: %s", name, err)
+		}
+	}
+	return nil
+}
+
+func TestAccAWSMwaaEnvironment_basic(t *testing.T) {
+	var environment mwaa.GetEnvironmentOutput
+
+	rName := acctest.RandomWithPrefix("tf-acc-test")
+	resourceName := "aws_mwaa_environment.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSMwaaEnvironmentDestroy,
```

Comment:
To prevent newer linter reports:

```suggestion
		PreCheck:     func() { testAccPreCheck(t) },
		ErrorCheck:   testAccErrorCheck(t, mwaa.EndpointsID),
		Providers:    testAccProviders,
		CheckDestroy: testAccCheckAWSMwaaEnvironmentDestroy,
```